### PR TITLE
Added support for adding stemming exceptions to the database

### PR DESCRIPTION
**Pull request by Autowinto**
Also reformatted existing code. Sorry about the many lines lmao

### CODE REVIEW

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java [hunk 1]
```diff
@@ -8,31 +8,41 @@
 
 public class ExceptionUtilities {
     public static void createException(String name) throws SQLException {
-        String query = String.format("INSERT INTO StemmingException (name) VALUES ('%s')", name);
-        ResultSet result = DBConnection.getConnection().createStatement().executeQuery(query);
-        result.close();
+        String query = String.format("INSERT INTO stemmingexceptions (exception) VALUES ('%s')", name);
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:11]: Please make prepared statements for queries and not just string interpolations.

**Review by OthelloEngineer — CHANGES_REQUESTED** [review]

**Comment by MagnusQuist** [comment]
"Do not use String.format() for that. Its role is quite different! String.format() cannot prevent SQL injection attacks; PreparedStatement can, unless its implementor did a really, really bad job."

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java (hunk 1 above)
Comment (Autowinto) [reply src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:11]: Pis også, you right. Coming right up

Comment (OthelloEngineer) [reply src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:11]: If you want it extracted out in private helper functions, see TwoWaySynonyms :)

**Comment by jakob753951** [comment]
i øvrigt mener jeg at getPooledConnection skal closes igen, så jeg ville bruge en try/with resources hvis jeg var jer

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java [hunk 2]
```diff
@@ -2,37 +2,63 @@
...
+        connection.close();
     }
+
     public static HashMap<Integer, String> getExceptions() throws SQLException {
-        String query = String.format("SELECT * FROM StemmingException");
-        ResultSet result = DBConnection.getConnection().createStatement().executeQuery(query);
-        result.close();
+        Connection connection = DBConnection.getPooledConnection();
+        String query = "SELECT * FROM stemmingexceptions";
+        ResultSet result = connection.prepareStatement(query).executeQuery();
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:24]: Is Connected closed here?

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java [hunk 3]
```diff
@@ -2,37 +2,63 @@
...
-    public static HashMap<Integer, String> findException(String name) throws SQLException {
-        String query = String.format("SELECT * FROM StemmingException WHERE name = '%s';", name);
-        ResultSet result = DBConnection.getConnection().createStatement().executeQuery(query);
-        result.close();
+
+    public static HashMap<Integer, String> getException(String name) throws SQLException {
+        Connection connection = DBConnection.getPooledConnection();
+        PreparedStatement statement = connection.prepareStatement("SELECT * FROM stemmingexceptions WHERE exception = ?");
+        statement.setString(1, name);
+        ResultSet result = statement.executeQuery();
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:32]: Is connection Closed here?

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java [hunk 4]
```diff
@@ -2,37 +2,63 @@
...
         return handleResult(result);
+
+    }
+
+    public static void updateException(String name, String newName) throws SQLException {
+        Connection connection = DBConnection.getPooledConnection();
+        PreparedStatement statement = connection.prepareStatement("UPDATE stemmingexceptions SET exception = ?" + "WHERE exception = ?");
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:39]: Where is this prepared statement closed? 
A preparedstatement, once queried, opens a cursor in the Database.

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java (hunk 2 above)
Comment (OthelloEngineer) [reply src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:24]: And is statement closed here
(Consider, that a resultset cannot be accessed without an open cursor)

File: src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java [hunk 5]
```diff
@@ -2,37 +2,63 @@
...
+        statement.setString(2, name);
+        statement.executeUpdate();
+        connection.close();
+    }
+
+    public static void deleteException(String name) throws SQLException {
+        Connection connection = DBConnection.getPooledConnection();
+        PreparedStatement statement = connection.prepareStatement("DELETE FROM stemmingexceptions " + "WHERE exception =  ?");
+        statement.setString(1, name);
+        statement.executeUpdate();
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/stemming/ExceptionUtilities.java:50]: plz close statement

**Review by OthelloEngineer — CHANGES_REQUESTED** [review]
Statments, resultset and connections need closure.

**Review by OthelloEngineer — APPROVED** [review]
LGTM

**Pull request merged**